core/sis-utility/src/main/java/org/apache/sis/measure/UnitRegistry.java

```diff
@@ -154,10 +154,9 @@ private static int filter(int existed, final SystemUnit<?> unit) {
      */
     static <Q extends Quantity<Q>> ConventionalUnit<Q> init(final ConventionalUnit<Q> unit) {
         assert !Units.initialized : unit;        // This assertion happens during Units initialization, but it is okay.
-        if (HARD_CODED.put(unit.getSymbol(), unit) == null) {
```

Comment:
This condition is one two separated lines because it represent two separated steps:

1. Register the unit symbol and check that there is no duplication.
2. If above step succeed, then register EPSG code and check that there is no duplication.

Admittedly we could have added comment for making those two steps more obvious. So instead than putting everything is a single `if` statement, which would hide those two steps, I think that an explanatory comment would be more appropriate.